pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -251,29 +256,27 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 		TpPairToMatchNum:     make(map[topologyPair]int, sizeHeuristic(len(allNodes), constraints)),
 	}
 
-	requiredSchedulingTerm := nodeaffinity.GetRequiredNodeAffinity(pod)
 	tpCountsByNode := make([]map[topologyPair]int, len(allNodes))
+	requiredNodeAffinity := nodeaffinity.GetRequiredNodeAffinity(pod)
 	processNode := func(i int) {
 		nodeInfo := allNodes[i]
 		node := nodeInfo.Node()
 		if node == nil {
 			klog.ErrorS(nil, "Node not found")
 			return
 		}
-		// In accordance to design, if NodeAffinity or NodeSelector is defined,
```

Comment:
IIUC, Jordan would like to literally have the same old code running if the feature is disabled (return at this point).

I also considered this initially, but the new code has the same behavior. Since it might not be so trivial to see that it's the same, please just check the feature gate here and return. Similarly, check the feature gate within the loop to skip the checks.

Reply:
done.